pkg/operator/ceph/cluster/watcher.go

```diff
@@ -53,62 +55,65 @@ func (c *clientCluster) onK8sNode(object runtime.Object) bool {
 	if !ok {
 		return false
 	}
+	// Check if the node have OSDs already present on it, reconcile if OSDs are not present
+	if alreadyPresentOSDs[node.Name] == "" {
+		// Get CephCluster
+		cluster := c.getCephCluster()
```

Comment:
Instead of calling `getCephCluster()` that only gets the first cluster, we really want to iterate over all the clusters
edit: Looking at it again, I'm not clear how this event is triggered and how it will cause the reconcile to be triggered on an individual cluster. Maybe we can keep it with a single cluster for now, and separately investigate how to trigger the reconcile if there are multiple clusters (or maybe it will already work)